### DISCUSSION TITLE
Tracing the source of a webhook

### DISCUSSION THREAD

**Discussion by breaker-of-stone** [community/community#75322] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I have configured two webhooks that fire on updates to the same repository, and lost the source/owner of one of them.  I can provide a sample payload from the lost webhook.

How can I track the lost one down and make changes ?

I've lost track due to multiple account / repo ownership changes over time.  It's not visible to me under my personal or org accounts.

**Comment by farid-teymouri** [comment]
If you've lost track of a webhook and it's not visible under your personal or organizational accounts, you may need to use the GitHub API to retrieve a list of webhooks for a specific repository. Here are the steps you can follow:

### Using GitHub API:

1-Generate Personal Access Token:

- Go to your GitHub account settings.
- Navigate to "Developer settings" -> "Personal access tokens."
- Generate a new token with the repo scop

2-Retrieve Webhooks Using cURL:
Open a terminal and use cURL to retrieve the list of webhooks for your repository. Replace `<owner>`, `<repo>`, and `<token>` with your repository owner, repository name, and personal access token, respectively.

```bash
curl -H "Authorization: token <token>" https://api.github.com/repos/<owner>/<repo>/hooks
```
This will return a JSON response containing information about all webhooks for the specified repository.

3-Identify the Lost Webhook:
Look through the JSON response to identify the webhook you are looking for. Each webhook will have a unique `id`, and you can check the config section to see if it matches the settings you expect.

4-Update or Delete the Webhook:

- Once you've identified the lost webhook, you can either update its settings or delete it if you no longer need it.
- To update a webhook, you can use the following cURL command:
```bash
curl -X PATCH -H "Authorization: token <token>" -d '{"config": {"<new-config>": "<new-value>"}}' https://api.github.com/repos/<owner>/<repo>/hooks/<webhook-id>
```

- To delete a webhook, you can use the following cURL command:
```bash
curl -X DELETE -H "Authorization: token <token>" https://api.github.com/repos/<owner>/<repo>/hooks/<webhook-id>
```

Make sure to replace placeholders like `<owner>`, `<repo>`, `<token>`, `<new-config>`, `<new-value>`, and `<webhook-id>` with your actual information.


### Using GitHub UI:
If you prefer a more visual approach, you can also use the GitHub UI to manage webhooks:

1. Go to your repository on GitHub.
2. Click on "Settings" and then navigate to the "Webhooks" section.
3. Review the list of webhooks and identify the one you are looking for.
4. Make changes or delete the webhook as needed.

These steps should help you track down and make changes to the lost webhook associated with your repository.

**Comment by breaker-of-stone** [comment]
Sorry, that only shows the one webhook that I can still access via the GUI, not the 'hidden' one.

Can I securely send you a sample deploy packet from the hidden webhook?

**Comment by ghostinhershell** [comment]
Hi @breaker-of-stone! The best place to go for answers to account-specific questions like this one is opening a ticket on our [Support page](https://support.github.com/contact). Please open a ticket with them :)

**Comment by breaker-of-stone** [comment, marked as answer]
Eventually traced to a developer workstation simulating a webhook - there was no 'hidden webhook' after all.
Whoops.
